Dockerfile

```diff
@@ -1,47 +1,58 @@
 # syntax = docker/dockerfile:1.0-experimental
 ARG VAGRANT_VERSION=2.2.19
 
-
-FROM ubuntu:bionic as base
-
-RUN apt update \
-    && apt install -y --no-install-recommends \
-        bash \
-        ca-certificates \
-        curl \
-        git \
-        gosu \
-        kmod \
-        libvirt-bin \
-        openssh-client \
-        qemu-utils \
-        rsync \
-    && rm -rf /var/lib/apt/lists \
+FROM debian:stable-slim as base
+
+RUN apt-get -y -qq update \
+    && apt-get -y --no-install-recommends install \
+      bash \
+      ca-certificates \
+      curl \
+      git \
+      gosu \
+      kmod \
+      libguestfs-tools \
+      libvirt0 \
+      libvirt-clients \
+      libvirt-dev \
+      libxml2-dev \
+      libxslt-dev \
+      openssh-client \
+      openssh-sftp-server \
+      qemu-system \
+      qemu-utils \
+      rsync \
+      ruby-dev \
+      zlib1g-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
     ;
 
 RUN mkdir /vagrant
 ENV VAGRANT_HOME /vagrant
 
 ARG VAGRANT_VERSION
 ENV VAGRANT_VERSION ${VAGRANT_VERSION}
+
+ARG DEFAULT_UID=1000
+ARG DEFAULT_USER=vagrant
+ARG DEFAULT_GROUP=users
+
 RUN set -e \
-    && curl https://releases.hashicorp.com/vagrant/${VAGRANT_VERSION}/vagrant_${VAGRANT_VERSION}_x86_64.deb -o vagrant.deb \
-    && apt update \
-    && apt install -y ./vagrant.deb \
-    && rm -rf /var/lib/apt/lists/* \
-    && rm -f vagrant.deb \
+    && apt-get -y -qq update \
+    && curl -sSL -o /tmp/vagrant.deb "https://releases.hashicorp.com/vagrant/${VAGRANT_VERSION}/vagrant_${VAGRANT_VERSION}_x86_64.deb" \
+    && apt-get install -y /tmp/vagrant.deb \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
+    && useradd -M --uid ${DEFAULT_UID} --gid ${DEFAULT_GROUP} ${DEFAULT_USER} \
     ;
 
-
 ENV VAGRANT_DEFAULT_PROVIDER=libvirt
 
 FROM base as build
 
-# allow caching of packages for build
-RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
-RUN sed -i '/deb-src/s/^# //' /etc/apt/sources.list
-RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt \
```

Comment:
The mount caches are intended to help speed up rebuilds, prefer if they are kept in place.